doc/reference/power_management/index.rst

```diff
@@ -416,7 +443,19 @@ Resume Device asynchronously API
    int pm_device_get(const struct device *dev);
 
 Marks the device as being used. This API will asynchronously
-bring the device to resume state. The API returns 0 on success.
+bring the device to resume state if it was suspended. If the device
+was already active, it just increments the device usage count.
+The API returns 0 on success.
+
+Device drivers can monitor this operation waiting on the device
+condition variable :code:`dev->pm->condvar`. e.g:
+
+.. code-block:: c
+
+   k_mutex_init(&request_mutex);
```

Comment:
This all is an implementation detail, neither application nor driver should be dealing with this directly, this should be implemented in the runtime device power management and the power subsystem.

Reply:
That is why I've introduced an API for this in https://github.com/zephyrproject-rtos/zephyr/pull/35315/commits/7d5f4d638f968db1911efad9637d068141668836

The problem is that this API is still not merged and what is documented here is for what we have in tree.
Note that prior these changes the way to do this was similar but waiting on k_poll_signal.

Now, if you are saying that the information about how a device should monitor an async request is not relevant then we have very different opinion. What I can do:

1) Keep it this way and update it in that other pr  (if this one is going to be merged before)
2) Change it and document future API waiting that pr to be merged
3) Remove pm_runtime documentation

Reply:
I suggest move this commit from this PR into the other PR and keep documentation close to the implementation.